### PR TITLE
Fix handling of getting and creating domains

### PR DESCRIPTION
**Pull request by Arachnid**

### CODE REVIEW

File: src/ensRegistry.ts [hunk 1]
```diff
@@ -87,12 +102,7 @@ function _handleNewResolver(event: NewResolverEvent, isMigrated:boolean): void {
   let id = event.params.resolver.toHexString().concat('-').concat(event.params.node.toHexString())
 
   let node = event.params.node.toHexString()
-  let domain = new Domain(node)
-  if(node == ROOT_NODE){
-    domain.owner = EMPTY_ADDRESS
-    domain.isMigrated = isMigrated
```
Comment (makoto) [inline src/ensRegistry.ts]: You can get rid of `isMigrated:boolean` from `_handleNewResolver` as `isMigrated` variable is no longer in use

File: src/ensRegistry.ts [hunk 2]
```diff
@@ -19,13 +19,34 @@ import {
...
 // Handler for NewOwner events
-function _handleNewOwner(event: NewOwnerEvent, isMigrated:boolean): void {
+function _handleNewOwner(event: NewOwnerEvent, isMigrated: boolean): void {
   let account = new Account(event.params.owner.toHexString())
   account.save()
 
   let subnode = crypto.keccak256(concat(event.params.node, event.params.label)).toHexString()
-  let domain = new Domain(subnode)
+  let domain = getDomain(subnode);
+  if(domain == null) {
```
Comment (makoto) [inline src/ensRegistry.ts]: You have another `if(domain.name == null)` at line 51 so you can add it there rather than having own statement

Comment (Arachnid) [reply src/ensRegistry.ts]: We're checking `if(domain == null)` here, not `domain.name`.

Comment (makoto) [reply src/ensRegistry.ts]: ah sorry

**Pull request merged**